tests/test_models.py

```diff
@@ -608,39 +620,77 @@ def test_sweep_coalescence_same_seed(self):
 
-    @unittest.skip("Not clear what's happening now")
     def test_sweep_start_time_complete(self):
         sweep_model = msprime.SweepGenicSelection(
-            reference_size=0.25, position=0.5, start_frequency=0.6,
-            end_frequency=0.7, alpha=0.9, dt=0.001)
+            reference_size=0.25, position=0.5, start_frequency=0.01,
+            end_frequency=0.99, alpha=0.9, dt=0.001)
         t_start = 0.1
         ts = msprime.simulate(
             10, Ne=0.25,
             recombination_rate=2,
             demographic_events=[
                 msprime.SimulationModelChange(t_start, sweep_model)],
-            num_labels=2, random_seed=2)
+            random_seed=2)
         self.assertTrue(all(tree.num_roots == 1 for tree in ts.trees()))
 
-    @unittest.skip("Can't get sweep to be incomplete")
     def test_sweep_start_time_incomplete(self):
         # Short sweep that doesn't make complete coalescence.
         sweep_model = msprime.SweepGenicSelection(
             reference_size=0.25, position=0.5, start_frequency=0.69,
-            end_frequency=0.7, alpha=1e-5, dt=1)
+            end_frequency=0.7, alpha=8, dt=0.001)
         t_start = 0.1
         ts = msprime.simulate(
             10, Ne=0.25,
             recombination_rate=2,
             demographic_events=[
                 msprime.SimulationModelChange(t_start, sweep_model)],
-            num_labels=2, random_seed=2)
+            random_seed=2)
+        self.assertTrue(any(tree.num_roots > 1 for tree in ts.trees()))
+
+    def test_sweep_model_change_time_complete(self):
+        # Short sweep that doesn't coalesce followed
+        # by Hudson phase to finish up coalescent
+        sweep_model = msprime.SweepGenicSelection(
+            reference_size=0.25, position=0.5, start_frequency=0.69,
+            end_frequency=0.7, alpha=1e-5, dt=1)
+        ts = msprime.simulate(
+            10, Ne=0.25,
+            recombination_rate=2,
+            demographic_events=[
+                msprime.SimulationModelChange(0, sweep_model),
+                msprime.SimulationModelChange(None, "hudson")],
+            random_seed=2)
+        self.assertTrue(all(tree.num_roots == 1 for tree in ts.trees()))
+        # Make sure that the Hudson phase did something.
+        ts = msprime.simulate(
+            10, Ne=0.25,
+            recombination_rate=2,
+            demographic_events=[
+                msprime.SimulationModelChange(0, sweep_model)],
+            random_seed=2)
         self.assertTrue(any(tree.num_roots > 1 for tree in ts.trees()))
+
+    def test_many_sweeps(self):
+        sweep_models = [
+            msprime.SweepGenicSelection(
+                reference_size=0.25, position=j, start_frequency=0.69,
+                end_frequency=0.7, alpha=1e-5, dt=0.1)
+            for j in range(10)]
+        ts = msprime.simulate(
+            10, Ne=0.25, length=10,
+            recombination_rate=0.2,
+            demographic_events=[
```

Comment:
nice. a cool thing to do next here would be to set a rate of sweeps, or any arbitrary model change, going back in time. i.e. we could imagine that sweeps occur in time according to a poisson point process

Reply:
Cool, I'll have a go at that. Lots of very fast sweeps, occurring at random times...

Reply:
it will be just like the original recurrent hitchhiking model!